htlcswitch/hop/payload.go

```diff
@@ -439,17 +537,15 @@ func validateBlindedRouteTypes(parsedTypes tlv.TypeMap,
 		}
 	}
 
-	for tlvType, tlvValue := range parsedTypes {
+	for tlvType, _ := range parsedTypes {
```

Comment:
This was modified to that validation works. Will leave a comment on Carla's PR and see if this change is good.

Looks like the `tlv` package returns a map from `tlv.Type` # to `nil` byte slices for [successfully parsed known types](https://github.com/lightningnetwork/lnd/blob/v0.15.4-beta/tlv/stream.go#L237) (ex:  `map[2:[] 4:[] 10:[]]`), so I think this check could be `if tlvValue == nil`. Then again, if we encounter an unknown "it's okay to be odd" type [we do store the raw bytes representing the value](https://github.com/lightningnetwork/lnd/blob/v0.15.4-beta/tlv/stream.go#L266) so this check may actually not be needed at all. If the `tlvType` is not in the map of `allowedTLVs` then we have an error!

Reply:
Should be resolved upstream but it was very late and very friday when I changed it, so lmk if it's resolved when you rebase (thoughts and prayers for that).